saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLMultiGraphDataModel.scala

```diff
@@ -36,19 +39,27 @@ class SRLMultiGraphDataModel(parseViewName: String = null, frameManager: SRLFram
 
   val tokens = node[Constituent]((x: Constituent) => x.getTextAnnotation.getCorpusId + ":" + x.getTextAnnotation.getId + ":" + x.getSpan)
 
+  // Edges
+
   val sentencesToStringTree = edge(sentences, stringTree)
   val sentencesToTokens = edge(sentences, tokens)
   val sentencesToRelations = edge(sentences, relations)
   val relationsToPredicates = edge(relations, predicates)
   val relationsToArguments = edge(relations, arguments)
 
+  // Sensors
+
   sentencesToTokens.addSensor(CommonSensors.textAnnotationToTokens _)
   sentencesToRelations.addSensor(textAnnotationToRelation _)
-  sentencesToRelations.addSensor(textAnnotationToRelationMatch _)
+
+  // TODO - Verify if we really need this matching sensor. This is very inefficient.
+  //  sentencesToRelations.addSensor(textAnnotationToRelationMatch _)
```

Comment:
@kordjamshidi Commenting out this matching sensor mitigates the data model population inefficiency. I have tried training models after commenting this out and training/testing works fine. Let me know if you have any concerns about this or any use-case that I didn't verify.

The root problem is still not fixed. The `matching` logic iterates over all possible instantiations of the edges while adding a single instance. [Source](https://github.com/CogComp/saul/blob/master/saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/edge/Edge.scala#L61)